src/v/net/conn_quota.cc

```diff
@@ -9,17 +9,22 @@
 
 #include "conn_quota.h"
 
+#include "bytes/oncore.h"
 #include "config/configuration.h"
 #include "config/validators.h"
 #include "hashing/xx.h"
 #include "rpc/logger.h"
 #include "seastar/core/coroutine.hh"
 #include "ssx/future-util.h"
+#include "vassert.h"
 #include "vlog.h"
 
+#include <seastar/core/smp.hh>
+
 namespace net {
 
 conn_quota::units::~units() noexcept {
+    oncore_debug_verify(_verify_shard);
```

Comment:
We can miss many cases with debug-only check. For example, all CDT CI run release build. I would suggest to vlog an error ("soft assert") in case we don't want to assert on prod. Maybe `oncore` can be extended accordingly?

Reply:
Good idea, I originally had it this way but dropped after switching to oncore. I've extended oncore.